### PR TITLE
Add AUTO_RPATH for py-lxml (#18293) and add version 4.5.2

### LINKED ISSUE #18293: Installation issue: py-lxml doesn't use the spack libxslt and libexslt libraries
It looks like py-lxml should have AUTO_RPATH set to true. Otherwise it picks up the OS versions of libxslt instead of the spack built versions. I added this to the package.py and the library dependencies were correct:

```
    def setup_build_environment(self, env):
        env.set('AUTO_RPATH', 'true')
```

### PR DESCRIPTION
**Pull request by ajw1980**
Fixes #18293

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lxml/package.py [hunk 1]
```diff
@@ -30,3 +31,6 @@ class PyLxml(PythonPackage):
     depends_on('py-html5lib', when='+html5', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lxml/package.py:31]: The libxml2 and libxslt dependencies should also be `type='link'` if we want to use Spack's compiler wrappers to locate and RPATH them. In that case, `AUTO_RPATH` may not even be needed. Can you try removing `AUTO_RPATH` and add `link` and see if this works?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-lxml/package.py (hunk 1 above)
Comment (ajw1980) [reply var/spack/repos/builtin/packages/py-lxml/package.py:31]: Seems like link works as well. I made the change.

**Comment by adamjstewart** [comment]
It looks like you accidentally closed the PR. Was that intentional?

**Comment by ajw1980** [comment]
Reopened

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**